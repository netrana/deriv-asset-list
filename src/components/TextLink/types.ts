export interface Props {
  className?: string;
  onClick: () => void;
}
