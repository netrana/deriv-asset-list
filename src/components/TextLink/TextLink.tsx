import React, { FC, MouseEventHandler } from "react";
import { Typography } from "antd";
import classNames from "classnames";

import { Props } from "./types";

import styles from "./textLink.module.scss";


const TextLink: FC<Props> = (props) => {
  const { children, onClick, className = ''} = props;
  const handleClick: MouseEventHandler = (event) => {
    event.stopPropagation();
    onClick();
  };

  return (
    <div
      className={classNames({
        [styles.textLink]: true,
        [className]: Boolean(className),
      })}
      onClick={handleClick}
    >
      <Typography.Text className={styles.text}>{children}</Typography.Text>
    </div>
  );
};

export default TextLink;
