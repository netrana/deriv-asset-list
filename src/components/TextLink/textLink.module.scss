@import 'styles/globals.scss';

.textLink {
  font-family: Poppins;
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 21px;
  color: rgba(0, 0, 0, 0.87);
  cursor: pointer;
}
