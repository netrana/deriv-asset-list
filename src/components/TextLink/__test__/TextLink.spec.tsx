import React from "react";
import render from "react-test-renderer";

import TextLink from "../TextLink";

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  FC: jest.fn(),
  MouseEventHandler: jest.fn(),
}));

jest.mock("antd", () => ({
  ...jest.requireActual("antd"),
  Typography: {
    ...jest.requireActual("antd").Typography,
    Text: "Text",
  },
}));

describe("TextLink component test", () => {
  describe("snapshot test", () => {
    it("should match snapshot", () => {
      const tree = render.create(<TextLink onClick={jest.fn} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
