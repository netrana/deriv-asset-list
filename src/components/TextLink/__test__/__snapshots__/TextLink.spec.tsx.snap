// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextLink component test snapshot test should match snapshot 1`] = `
<div
  className="textLink"
  onClick={[Function]}
>
  <Text
    className="text"
  />
</div>
`;
